### ISSUE TITLE
Export with template

### ISSUE THREAD

**Issue by duckinator** [duckinator/STAYFOCUSED#9]
Given:
- A series of tasks, with tags (blocked by #8)
- A per-tag template
- A per-tag file path template

I want to be able to click "Export" and get all of the current information saved to a file.

An example template might be:

```
Ellen Dash: {{ time }} {{ task }}
{{ notes }}

```

And an example file path template might be:

```
/home/puppy/work/{{ tag }}/%y-%m-update.txt
```

If you have tasks with tags "Foo" and "Bar", this means if I exported things today I'd get:
- `/home/puppy/work/Foo/2024-09-update.txt` containing all things tagged `Foo`
- `/home/puppy/work/Bar/2024-09-update.txt` containing all things tagged `Bar`

**Comment by duckinator** [comment]
brain dump:

> and notably, the way i'm thinking of making STAYFOCUSED work is:
> 
> - each task can have tags assigned to it
> - each tag can be assigned a template file
> - you can perform an export-to-file of everything associate with that template file
> 
> and the export-to-file would apply the template to each task with that tag, ensure there's a blank line between each one, and save it to a specified file
> 
> so you could have e.g. a 'Fractal' tag with this template:
> 
> ```
> {{ elapsed_time }} {{ description }}
> {{ notes }}
> ```
> 
> and the result might look something like:
> 
> ```
> 3h Some example task
> Some notes about "Some example task"
> 
> 2.25h Another task
> Notes about "another task"
> ```
> 
> and maybe an "export time log" option that just uses a template of `{{ elapsed_time }} {{ description }}`, so with those same tasks you'd get:
> 
> ```
> 3h Some example tasks
> 2.25h Another task
> ```
> 
> if whatever templating thing i use has filters, i'll probably ensure there's one for padding strings with spaces to a certain length, so you could e.g. make sure elapsed_time is always formatted with enough spaces to fill 6 characters (enough to fit up to 99.99h -- if you need more than that on one task you should probably break the task up, and you just don't need more than two decimal places of precision, and i'd argue you only need that for .25 and .75).
